src/tree/treemath.rs

```diff
@@ -96,71 +97,72 @@ pub(crate) fn sibling(index: NodeIndex, size: LeafIndex) -> Result<NodeIndex, Tr
     }
 }
 
-// Ordered from leaf to root
-// Includes neither leaf nor root
-pub(crate) fn dirpath(index: NodeIndex, size: LeafIndex) -> Result<Vec<NodeIndex>, TreeMathError> {
-    let r = root(size);
-    if index == r {
-        return Ok(vec![]);
-    }
-
-    let mut d = vec![];
-    let mut x = parent(index, size)?;
-    while x != r {
-        d.push(x);
-        x = parent(x, size)?;
-    }
-    Ok(d)
-}
-
-// Ordered from leaf to root
-// Includes leaf and root
-pub(crate) fn dirpath_long(
-    index: NodeIndex,
+/// Direct path from a leaf node to the root.
+/// Does not include the leaf node but includes the root.
+pub(crate) fn leaf_direct_path(
+    leaf_index: LeafIndex,
     size: LeafIndex,
 ) -> Result<Vec<NodeIndex>, TreeMathError> {
+    let node_index = NodeIndex::from(leaf_index);
     let r = root(size);
-    if index == r {
+    if node_index == r {
         return Ok(vec![r]);
     }
 
-    let mut x = index;
-    let mut d = vec![index];
+    let mut d = vec![];
+    let mut x = node_index;
     while x != r {
         x = parent(x, size)?;
         d.push(x);
     }
     Ok(d)
 }
 
-// Ordered from leaf to root
-// Includes root but not leaf
-pub(crate) fn direct_path_root(
-    index: NodeIndex,
+/// Direct path from a parent node to the root.
+/// Includes the parent node and the root.
+/// Returns an error if the `index` is not a parent node.
+pub(crate) fn parent_direct_path(
```

Comment:
I'm not sure this is a good idea. `direct_path` is well defined in the spec not to include the node it starts from. If we want to include the starting node itself, we should either call the function differently, or do it ad-hoc when we need it.

Reply:
I don't really care about the spec here but we should be consistent in the naming and semantics.

Reply:
I thought we agreed (and I think it was also your suggestion) that it is ok to deviate from the spec when it makes sense to improve the quality of the implementation. I believe this is one of those instances.
If we do more refactoring around the binary tree, we might have to change this anyway, but for now it's the most generic way and it gives additional type safety.

Reply:
Can we change the name then or something? The actual issue is still here.

Reply:
I'm not sure I follow what the issue is. Why is the name a problem?

For context: Currently the implementation uses direct path in two instances: when starting with a leaf and when starting with a parent node. I named the two variants of direct path to reflect the use case.

Reply:
Right, the issue is that `parent_direct_path` has a different result than `leaf_direct_path` because it includes the starting node. Having so similar names this behaviour is unexpected. So *something* should change.

Reply:
Would it help if I renamed it to `partial_direct_path()`?

For context: The risk of confusion is low, because the two functions have different signatures as well.

Reply:
I'm not worried about confusion but about assumptions on what the function does. The easiest would be to always either include the starting node or not.

Reply:
I think what the functions do is the most generic we can get and the documentation is also accurate, which leaves us with the name:

I suggested `partial_direct_path()`, `inner_direct_path()` could also work. If those are not good enough I think it's best if you make a suggestion so that we can move forward.

Reply:
I'm not excited about the names but don't have a better idea. I'll approve and let you decide :)